code/modules/vehicles/unmanned/remote_control.dm

```diff
@@ -0,0 +1,48 @@
+/datum/component/remote_control
+	var/mob/user
+	var/atom/target
+	var/is_controlling = FALSE
+
+
+/datum/component/remote_control/Initialize(atom/target)
+	. = ..()
+	src.target = target
+
+	RegisterSignal(parent, COMSIG_REMOTE_CONTROL_TOGGLE, .proc/toggle_remote_control)
+
+/datum/component/remote_control/Destroy()
+	return ..()
+
+/datum/component/remote_control/proc/on_user_clickon(datum/source, atom/A, params)
+	if(A==parent)
+		return ..()
+	if(istype(target, /obj/vehicle/unmanned))
```

Comment:
This check should happen in `Initialize()`, not after every click.
You can save the `on_user_clickon` proc path to a variable on initialize, and register it during `remote_control_on`